File: rvsf/public/js/supplier_quotation.js
frappe.ui.form.on("Supplier Quotation", {
    onload_post_render(frm) {
        if (frm.doc.custom_purchase_lead) {
            delete frm.custom_make_buttons["Quotation"];
            frm.remove_custom_button(__("Quotation"), __("Create"));
        }
    },
    after_workflow_action(frm) {
        if (frm.doc.workflow_state === "Rejected") {
            console.log("rejected")
            let d = new frappe.ui.Dialog({
                title: "Reason for Rejection",
                fields: [
                    {
                        label: "Reason",
                        fieldname: "reason",
                        fieldtype: "Small Text",
                        reqd: 1
                    }
                ],
                primary_action_label: "Submit",
                primary_action(values) {
                    frm.set_value(
                        "custom_reason_for_rejection",
                        values.reason
                    );
                    d.hide();
                    frappe.show_alert({
                        message: __("Rejection reason added"),
                        indicator: "green"
                    }, 5);
                }
            });
            d.show();
        }
        frappe.call({
            method: "rvsf.rvsf.events.supplier_quotation.update_supplier_quotation_status",
            args: {
                supplier_quotation:frm.doc.name
            }
        })
    }
})

File: rvsf/rvsf/events/supplier_quotation.py
import frappe


@frappe.whitelist()
def update_supplier_quotation_status(supplier_quotation):
    doc = frappe.get_doc("Supplier Quotation", supplier_quotation)
    purchase_lead = frappe.get_doc("Purchase Lead", doc.custom_purchase_lead)
    if not purchase_lead:
        return
    if doc.workflow_state == "Accepted":
        purchase_lead.db_set("status", "Accepted")
    elif doc.workflow_state == "Rejected":
        purchase_lead.db_set("status", "Rejected")
    elif doc.workflow_state == "Revised":
        purchase_lead.db_set("status", "Revised")
    elif doc.workflow_state == "Sent":
        purchase_lead.db_set("status", "Quotation Sent")
    elif doc.workflow_state == "Cancelled":
        purchase_lead.db_set("status", "Open")